notify/drivers/salesforce/salesforce.py

```diff
@@ -0,0 +1,390 @@
+            LOG.debug('Upate status code: {} '.format(u.status_code))
+
+            feeditem_data = {
+                'ParentId':    ExistingCaseId,
+                'Visibility': 'AllUsers',
+                'Body':        json_f(feeditem_body)
+            }
+
+            LOG.debug('FeedItem Data: {}'.format(json_f(feeditem_data)))
+            add_feed_item = sfdc_client.create_feeditem(feeditem_data)
+            LOG.debug("Add FeedItem status code: {} \n Add FeedItem reply: {} "
+                      "".format(add_feed_item.status_code, add_feed_item.text))
+            return True
+        # Else If Case did not exist before and was just created
+        elif (new_case.status_code == 201):
+            LOG.debug('Case was just created')
+            # Add commnet, because Case head should conains
+            # LAST data  overriden on any update
+            CaseId = new_case.json()['id']
+            feeditem_data = {
+                'ParentId':   CaseId,
+                'Visibility': 'AllUsers',
+                'Body': json_f(feeditem_body),
+            }
+            LOG.debug('FeedItem Data: {}'.format(json_f(feeditem_data)))
+            add_feed_item = sfdc_client.create_feeditem(feeditem_data)
+            LOG.debug("Add FeedItem status code: {} \n Add FeedItem reply: {} "
+                      "".format(add_feed_item.status_code, add_feed_item.text))
+            return True
+        else:
+            LOG.debug("Unexpected error: Case was not created (code != 201) "
+                      "and Case does not exist (code != 400)")
+            return False
+
+    except requests.exceptions.ConnectionError as e:
+        LOG.debug(e)
+        LOG.debug('Unexpected error: Case was not created: Connection error.')
+        return False
+
+
+class OAuth2(object):
+    def __init__(self,
+                 client_id,
+                 client_secret,
+                 username,
+                 password,
+                 auth_url=None,
+                 organizationId=None):
+
+        if not auth_url:
+            auth_url = 'https://login.salesforce.com'
+
+        self.auth_url = auth_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.username = username
+        self.password = password
+        self.organizationId = organizationId
+
+    def getUniqueElementValueFromXmlString(self, xmlString, elementName):
+        """Extracts an element value from an XML string.
+
+        For example, invoking
+        getUniqueElementValueFromXmlString('
+           <?xml version="1.0" encoding="UTF-8"?><foo>bar</foo>', 'foo'
+        )
+        should return the value 'bar'.
+        """
+        xmlStringAsDom = xml.dom.minidom.parseString(xmlString)
+        elementsByName = xmlStringAsDom.getElementsByTagName(elementName)
+        elementValue = None
+        if len(elementsByName) > 0:
+            toXML = elementsByName[0].toxml()
+            r1 = '<' + elementName + '>'
```

Comment:
Will this work if tag has some attributes? (example: \<foo someattr="spam"\>bar\</foo\>) ?

I would recommend to use native way (no regex):

    from xml.dom import minidom

    def get_first_node_value(xml_str, node_name):
        els = minidom.parseString(xml_str).getElementsByTagName(node_name)
        if els:
            return els[0].firstChild.nodeValue
        return None